OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -128,13 +140,21 @@ public override bool Tick(Actor self)
 
 			// If all valid weapons have depleted their ammo and Rearmable trait exists, return to RearmActor to reload
 			// and resume the activity after reloading if AbortOnResupply is set to 'false'
-			if (rearmable != null && !useLastVisibleTarget && attackAircraft.Armaments.All(x => x.IsTraitPaused || !x.Weapon.IsValidAgainst(target, self.World, self)))
+			// Fix the weapon validity check
+			if (rearmable != null && !useLastVisibleTarget &&
+				attackAircraft.Armaments.All(x =>
+					x.IsTraitPaused ||
+					(rearmable != null &&
+						(self.TraitsImplementing<AmmoPool>().FirstOrDefault() == null ||
+						 !self.TraitsImplementing<AmmoPool>().FirstOrDefault().HasAmmo)) ||
+					!x.Weapon.IsValidAgainst(target, self.World, self)))
```

Comment:
It's assumed that if all armamanets are paused, it means we are out of ammo. This is true in base mods so this change isn't needed